cloudinit/sources/DataSourceVMware.py

```diff
@@ -0,0 +1,816 @@
+                by_ipv4[key] = val
+
+        if af_inet6:
+            for ip_info in af_inet6:
+                key = ip_info["addr"]
+                if not is_valid_ip_addr(key):
+                    continue
+                val = copy.deepcopy(ip_info)
+                del val["addr"]
+                if mac:
+                    val["mac"] = mac
+                by_ipv6[key] = val
+
+    return host_info
+
+
+def wait_on_network(metadata):
+    # Determine whether we need to wait on the network coming online.
+    wait_on_ipv4 = False
+    wait_on_ipv6 = False
+    if WAIT_ON_NETWORK in metadata:
+        wait_on_network = metadata[WAIT_ON_NETWORK]
+        if WAIT_ON_NETWORK_IPV4 in wait_on_network:
+            wait_on_ipv4_val = wait_on_network[WAIT_ON_NETWORK_IPV4]
+            if isinstance(wait_on_ipv4_val, bool):
+                wait_on_ipv4 = wait_on_ipv4_val
+            else:
+                wait_on_ipv4 = bool(strtobool(wait_on_ipv4_val))
+        if WAIT_ON_NETWORK_IPV6 in wait_on_network:
+            wait_on_ipv6_val = wait_on_network[WAIT_ON_NETWORK_IPV6]
+            if isinstance(wait_on_ipv6_val, bool):
+                wait_on_ipv6 = wait_on_ipv6_val
+            else:
+                wait_on_ipv6 = bool(strtobool(wait_on_ipv6_val))
+
+    # Get information about the host.
+    host_info = None
+    while host_info is None:
+        host_info = get_host_info()
+        if wait_on_ipv4:
+            ipv4_ready = False
+            if "network" in host_info:
+                if "interfaces" in host_info["network"]:
+                    if "by-ipv4" in host_info["network"]["interfaces"]:
+                        if (
+                            len(host_info["network"]["interfaces"]["by-ipv4"])
+                            > 0
+                        ):
+                            ipv4_ready = True
+            if not ipv4_ready:
+                LOG.info("ipv4 not ready")
+                host_info = None
+        if wait_on_ipv6:
+            ipv6_ready = False
+            if "network" in host_info:
+                if "interfaces" in host_info["network"]:
+                    if "by-ipv6" in host_info["network"]["interfaces"]:
+                        if (
+                            len(host_info["network"]["interfaces"]["by-ipv6"])
+                            > 0
+                        ):
+                            ipv6_ready = True
+            if not ipv6_ready:
+                LOG.info("ipv6 not ready")
+                host_info = None
+        if host_info is None:
+            LOG.info("waiting on network")
+            time.sleep(1)
+
+    return host_info
+
+
+def get_data_access_method():
+    if os.environ.get(VMX_GUESTINFO, ""):
+        return VMX_GUESTINFO
+    if VMWARE_RPCTOOL:
+        return VMWARE_RPCTOOL
+    return None
+
+
+_MERGE_STRATEGY_ENV_VAR = "CLOUD_INIT_VMWARE_GUEST_INFO_MERGE_STRATEGY"
+_MERGE_STRATEGY_DEEPMERGE = "deepmerge"
+
+
+def merge_dicts(a, b):
```

Comment:
I haven't grokked all the merging code here, but would `util.mergemanydict` work as a replacement?

Reply:
Great catch, yes it can!